### ISSUE TITLE
Evaluation error: a ‘armv7l-linux’ is required to build (...), but I am a ‘x86_64-linux’

### ISSUE THREAD

**Issue by wizeman** [NixOS/hydra#479]
I have an `x86_64-linux` Hydra server with an `armv7l-linux` build slave, both running NixOS stable (with a few seemingly unrelated additional backports and fixes), and I'm trying to build an `armv7l-linux` NixOS configuration in a build job. This used to work before, but since a couple of weeks ago (having changed nothing except updating the NixOS stable channel), I'm getting this error during evaluation:

```
hydra-eval-jobs returned exit code 1:
warning: in configuration file '/etc/nix/nix.conf': unknown setting 'binary-cache-files'
warning: in configuration file '/etc/nix/nix.conf': unknown setting 'binary-caches-files'
warning: in configuration file '/etc/nix/nix.conf': unknown setting 'use-binary-caches'
building path(s) ‘/nix/store/4qzrwys0wyzk58mz3hx7j997544xrlnk-tvrasp-src’
error: a ‘armv7l-linux’ is required to build ‘/nix/store/1jqnzj29nny7shc5qr5h4yzjxh93sl7i-tvrasp-src.drv’, but I am a ‘x86_64-linux’
(use ‘--show-trace’ to show detailed location information)
```

I have checked that the build slave is being correctly recognized in the `Machine status` page, along with the correct architecture.

Note that the derivation mentioned above is the unpacked source which is being used in an import-from-derivation, but it may just be that this is the very first derivation that the Hydra server is trying to build for that build job.

If I run build the configuration using `nixops deploy` from another machine (but using the same build slave) it builds and deploys fine.

Any ideas?

**Comment by domenkozar** [comment]
Yeah, nixpkgs shouldn't use import-from-derivation, this is disallowed in newer Nix, but I'm not sure it's deployed to hydra.nixos.org yet

**Comment by cleverca22** [comment]
i also ran into this issue with `haskellPackages.callCabal2nix`, hydra doesn't allow build slaves at eval time, and `haskellPackages.callCabal2nix` tried to run the darwin version of `cabal2nix`, because a darwin instance of nixpkgs isnt capable of mixing linux programs into its IFD things
